File: python/meteroid/models/update_metric_request.py
# this file is @generated
import dataclasses
import typing as t

from ..serialization import BaseModel
from .metric_filter import MetricFilter
from .metric_segmentation_matrix import MetricSegmentationMatrix
from .unit_conversion import UnitConversion


@dataclasses.dataclass
class UpdateMetricRequest(BaseModel):
    description: t.Optional[str] = None

    filters: t.Optional[t.List[MetricFilter]] = None
    """Absent = leave filters untouched; present (even empty) = replace them."""

    name: t.Optional[str] = None

    segmentation_matrix: t.Optional[MetricSegmentationMatrix] = None

    unit_conversion: t.Optional[UnitConversion] = None

File: go/update_metric_request.go
// this file is @generated
package meteroid

type UpdateMetricRequest struct {
	Description *string `json:"description,omitempty"`

	// Absent = leave filters untouched; present (even empty) = replace them.
	Filters []MetricFilter `json:"filters,omitempty"`

	Name *string `json:"name,omitempty"`

	SegmentationMatrix *MetricSegmentationMatrix `json:"segmentation_matrix,omitempty"`

	UnitConversion *UnitConversion `json:"unit_conversion,omitempty"`
}
